src/appleseed/renderer/modeling/frame/frame.cpp

```diff
@@ -776,6 +804,30 @@ bool Frame::write_main_and_aov_images() const
         }
     }
 
+    if (!aov_images().empty())
+    {
+        const bf::path boost_file_path(filepath);
+        const bf::path directory = boost_file_path.parent_path();
+        const string base_file_name = boost_file_path.stem().string();
+        const string extension = boost_file_path.extension().string();
+
+        // Extra AOVs
+        for (size_t i = 0, e = aov_images().size(); i < e; ++i)
```

Comment:
This fragment of code is repeated. Refactor into a function in an unnamed namespace if possible, 
if not, refactor to a private method of the frame.